### DISCUSSION TITLE
Is there a way to use yazi in nvim without a floating window

### DISCUSSION THREAD

**Discussion by JumpyLionnn** [mikavilpas/yazi.nvim#303] [Q&A] [answered]
Is there a way to use yazi in nvim without a floating window so when i close nvim when yazi is active it wont redirect me to the open buffer?
for example when i use `nvim .` to open the current directory and the i press `q` to quit i get a blank buffer which i need to close as well with `:q`.
i dont really like the floating window, is there a way to disable it?

**Comment by mikavilpas** [comment, marked as answer]
Right now it's required. I started creating a proof of concept where yazi could be opened in the current window (meaning, split or buffer), so it might be an option in the future.

However, closing Neovim when yazi is closed might be a disruptive default for some users.

Still, it should be possible to do what I think it is you are asking. I would suggest looking into the advanced configuration section of the readme. You can find a section showing some hooks, and I think you could implement the following logic in your configuration:

- when yazi is closed, check if there are other open buffers in Neovim
- if there are no open buffers, execute `:qa` to quit

**Reply by JumpyLionnn** [reply]
Thank you!
This is what I ended up with after a bit of researching
```lua
yazi_closed_successfully = function(chosen_file, _, _)
    local bufs = vim.api.nvim_list_bufs();
    local buffers_empty = true;
    for _, buf in ipairs(bufs) do
        local lines = vim.api.nvim_buf_get_lines(buf, 0, -1, false);
        if #lines > 1 or lines[1] ~= "" then
            buffers_empty = false;
            break;
        end
    end
    if chosen_file == nil and buffers_empty then
        vim.cmd("qa")
    end
end
```
This is not the best but it will do the job for me.
